src/schema/date_time_options.rs

```diff
@@ -0,0 +1,288 @@
+use std::ops::BitOr;
+
+use serde::{Deserialize, Serialize};
+
+use super::Cardinality;
+use crate::schema::flags::{FastFlag, IndexedFlag, SchemaFlagList, StoredFlag};
+
+/// DateTime Precision
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum DatePrecision {
+    /// Seconds precision
+    Seconds,
+    /// Milli-seconds precision.
+    Milliseconds,
+    /// Micro-seconds precision.
+    Microseconds,
+}
+
+impl Default for DatePrecision {
+    fn default() -> Self {
+        DatePrecision::Seconds
+    }
+}
+
+/// Defines how DateTime field should be handled by tantivy.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DateOptions {
+    indexed: bool,
+    // This boolean has no effect if the field is not marked as indexed true.
+    fieldnorms: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    fast: Option<Cardinality>,
+    stored: bool,
+    // Internal storage precision, used to optimize storage
+    // compression on fast fields.
+    #[serde(default)]
+    precision: DatePrecision,
+}
+
+impl Default for DateOptions {
+    fn default() -> Self {
+        Self {
```

Comment:
You have clippy warnings here.

Reply:
used Default, shorter & the recommended Clippy way